plugins/woocommerce/src/Internal/Admin/BlockTemplates/AbstractBlockTemplate.php

```diff
@@ -131,4 +131,19 @@ function( BlockInterface $block ) {
 
 		return $inner_blocks_formatted_template;
 	}
+
+	/**
+	 * Get the template as JSON like array.
+	 *
+	 * @return array The JSON.
+	 */
+	public function to_json(): array {
```

Comment:
This method should be on the `Automattic\WooCommerce\Admin\BlockTemplates\ContainerInterface` so that we can call this for any template, not just those that choose to use `AbstractBlockTemplate` as a base.

And, there should be some unit tests added for this method in the `Automattic\WooCommerce\Tests\Internal\Admin\BlockTemplates\BlockTemplateTest` class.

Reply:
Well I think it is ok if we want that function to be implemented by all subclasses which means that if I do it then I will have to add the function implementation in each subclass (also to not template ones), that was the reason why I just added it to the `AbstractBlockTemplate`. I will add the function definition to the `BlockTemplateInterface` instead so I can contain a little the scope of this PR. We should create an issue to improve this in the future. https://github.com/woocommerce/woocommerce/pull/41823/commits/318caa901b36af7eed7ac1219014f05d8ee0d509

Reply:
I agree, this is reasonable for now.